design/authenticators/authn_k8s/inject_client_cert_solution_design.md

```diff
@@ -0,0 +1,255 @@
+[//]: # "Elaborate on whether your solution will affect the product's performance, and how"
+The performance of the "inject-client-cert" request will not be affected as the Websocket is still being closed immediately after we inject the cert.
+
+### Security
+
+[//]: # "Are there any security issues with your solution? Even if you mentioned them somewhere in the doc it may be convenient for the security architect review to have them centralized here"
+
+1. Is it ok to pass the base64 encoded cert in the URL? Is it less secure than sending it in the STDIN?
+   1. The answer is yes and thus we are not going with option 2.
+
+## Affected Components
+
+[//]: # "Address the components that will be affected by your solution [Conjur, DAP, clients, integrations, etc.]"
+- Conjur
+- DAP
+- conjur-authn-k8s-client
+- seed-fetcher
+- secrets-provider-for-k8s
+
+## Cross-team dependencies
+[//]: # "Elaborate about your solution's requirements or needs from other teams"
+We would like to get a review on the solution from @cyberark/conjur-core-team  and @cyberark/community-and-integrations-team. There are no requirements from  the Infra team.
+
+## Test Plan
+[//]: # "Fill in the table below to depict the tests that should run to validate your solution"
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+The current integration tests of the "inject-client-cert" request are verifying that the cert is installed but they have 10 retries of 2 seconds each. If we decide to make the request atomic (i.e return 200 OK upon cert injection) then we should change this behaviour to have no retries.
+
+| **Given**                                                    | **When**                                | **Then**                                                     | **Comment**                                                  |
+| ------------------------------------------------------------ | --------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| An authn-client container with permissions on the "ssl" directory | An "inject-client-cert" request is sent | - The response code is 200 OK<br />- The cert is in the container |                                                              |
+| An authn-client container without permissions on the "ssl" directory | An "inject-client-cert" request is sent | - The response code is 401<br />- The error is written to the log |                                                              |
+| A service-account that has no sufficient permissions         | An "inject-client-cert" request is sent | - The response code is 401<br />- The error is written to the log | This test is different than the one above because the error type is different. This request will fail today as well while the previous one would have succeeded (although he cert wouldn't be installed). |
+
```

Comment:
do we have a good sense of the errors that will happen on the server side when the `kubectl exec` command is run after it's changed to a bash script?

Reply:
the server side will not have the errors as the socket is closed. the only errors that will be present in the server are handshake errors (same as today)